pyflakes/checker.py

```diff
@@ -1252,6 +1258,19 @@ def checkReturnWithArgumentInsideGenerator():
 
         self.deferFunction(runFunction)
 
+    def ARGUMENTS(self, node):
+        parent_function = self.getAncestor(node, (ast.FunctionDef, ast.Lambda))
```

Comment:
if you create a `getScopeNode` function, which gets the innermost functiondef or lambda or classdef or module , it should be the same result as what you want.

Your new `getScopeNode` can still be implemented as `self.getAncestor(node, (ast.FunctionDef, ast.Lambda, ast.ClassDef, ast.Module))` etc.  The wrapper `getScopeNode` just reduces the repetition, and supports any new scope which has args.

And this line should only occur in `if PY2:` as it is unneeded for Python 3